### PR TITLE
docs: jwe-decrypt secret length must be 32 chars

### PR DESCRIPTION
**Pull request by Vacant2333**
### Description

Fixes # [(issue 10825)](https://github.com/apache/apisix/issues/10825)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by Vacant2333** [comment]
@kayx23 @shreemaan-abhishek can u help me take a look~

File: docs/en/latest/plugins/jwe-decrypt.md [hunk 1]
```diff
@@ -38,11 +38,11 @@ This Plugin adds an endpoint `/apisix/plugin/jwe/encrypt` for JWE encryption. Fo
...
+| Name          | Type    | Required                                              | Default | Valid values                | Description                                                                                                                              |
+|---------------|---------|-------------------------------------------------------|---------|-----------------------------|------------------------------------------------------------------------------------------------------------------------------------------|
+| key           | string  | True                                                  |         |                             | Unique key for a Consumer.                                                                                                               |
+| secret        | string  | True                                                 |         |                             | The decryption key. The key could be saved in a secret manager using the [Secret](../terminology/secret.md) resource. (Must be 32 chars) |
```
Comment (kayx23) [inline docs/en/latest/plugins/jwe-decrypt.md:44]:
```suggestion
| secret        | string  | True                                                 |         |                             | The decryption key. Must be 32 characters. The key could be saved in a secret manager using the [Secret](../terminology/secret.md) resource. |
```

File: docs/zh/latest/plugins/jwe-decrypt.md [hunk 2]
```diff
@@ -38,11 +38,11 @@ description: 本文档包含了关于 APISIX jwe-decrypt 插件的相关信息
...
 
-| 名称          | 类型      | 必选项   | 默认值   | 有效值 | 描述                                                                   |
-|---------------|---------|-------|-------|-----|----------------------------------------------------------------------|
-| key           | string  | True  |       |     | Consumer 的唯一 key                                                     |
-| secret        | string  | True  |       |     | 解密密钥。秘钥可以使用 [Secret](../terminology/secret.md) 资源保存在密钥管理服务中（最小 32 位） |
-| is_base64_encoded | boolean | False | false |     | 如果密钥是 Base64 编码，则需要配置为 `true`                                        |
+| 名称          | 类型      | 必选项   | 默认值   | 有效值 | 描述                                                                |
+|---------------|---------|-------|-------|-----|-------------------------------------------------------------------|
+| key           | string  | True  |       |     | Consumer 的唯一 key                                                  |
+| secret        | string  | True  |       |     | 解密密钥。秘钥可以使用 [Secret](../terminology/secret.md) 资源保存在密钥管理服务中（32 位） |
```
Comment (kayx23) [inline docs/zh/latest/plugins/jwe-decrypt.md:44]:
```suggestion
| secret        | string  | True  |       |     | 解密密钥，必须为 32 位。秘钥可以使用 [Secret](../terminology/secret.md) 资源保存在密钥管理服务中  |
```

**Comment by Vacant2333** [comment]
i need time to confirm is the solution right, close the pr now

**Comment by Vacant2333** [comment]
i have remote the `max-length` of the scheme, cause if user set `is_base64_encoded` will get some problem, can u help me review and merge @shreemaan-abhishek  @kayx23

**Comment by kayx23** [comment]
> i have remote the `max-length` of the scheme

You mean remove? And where is it removed?

**Comment by Vacant2333** [comment]
> > i have remote the `max-length` of the scheme
> 
> You mean remove? And where is it removed?

https://github.com/apache/apisix/pull/10883/commits/f2890ef050deb776744605b42c6cd0eb8a56c688#diff-6529ea6d6dfb2a162df8b4ef89a760c9acba198e139fba0faa58aed1e18dac08L50 here, In the first commit, I wanted to limit its length to the exact 32 bytes, but ci failed, because we have the requirement of base64, so I shouldn't have limited the length in scheme. I removed it in the second commit.

**Comment by shreemaan-abhishek** [comment]
@Vacant2333 you must also update the schema, shouldn't you? And is there any spec/rfc that supports your claim?

**Comment by Vacant2333** [comment]
> @Vacant2333 you must also update the schema, shouldn't you? And is there any spec/rfc that supports your claim?

no, we cant limit the schemea, if use set `is_base64_encoded` to `true`, the secret will longger than 32 chars

**Review by Revolyssup — APPROVED** [review]
LGTM

**Review by shreemaan-abhishek — APPROVED** [review]

**Comment by kayx23** [comment]
> > @Vacant2333 you must also update the schema, shouldn't you? And is there any spec/rfc that supports your claim?
> 
> no, we cant limit the schemea, if use set `is_base64_encoded` to `true`, the secret will longger than 32 chars. u can check this testcase, the secret after base64_encode will be longer than 32chars https://github.com/apache/apisix/blob/master/t/plugin/jwe-decrypt.t#L348

@Vacant2333 In that case, why is this PR still saying the secret must be 32 chars, instead of explaining more of what you explain above?

**Comment by Vacant2333** [comment]
> > > @Vacant2333 you must also update the schema, shouldn't you? And is there any spec/rfc that supports your claim?
> > 
> > 
> > no, we cant limit the schemea, if use set `is_base64_encoded` to `true`, the secret will longger than 32 chars. u can check this testcase, the secret after base64_encode will be longer than 32chars https://github.com/apache/apisix/blob/master/t/plugin/jwe-decrypt.t#L348
> 
> @Vacant2333 In that case, why is this PR still saying the secret must be 32 chars, instead of explaining more of what you explain above?

yes, u r right! i will raise a pr to mention these infomations, and it will check the secret length when APISIX running

**Pull request merged**